### PR TITLE
enhance array_to_lod_tensor_op lod_tensor_to_array_op errors informaiton

### PR DESCRIPTION
**Pull request by yongqiangma**
### PR types

Others
### PR changes

OPs
### Describe

enhance array_to_lod_tensor_op lod_tensor_to_array_op errors informaiton

### CODE REVIEW

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 1]
```diff
@@ -48,7 +48,8 @@ struct ArrayToLoDFunctor : public boost::static_visitor<void> {
 #ifdef PADDLE_WITH_CUDA
       Apply(static_cast<platform::CUDADeviceContext *>(pool.Get(place)));
 #else
-      PADDLE_THROW("Fluid is not compiled with CUDA");
+      PADDLE_THROW(
+          platform::errors::Unavailable("Fluid is not compiled with CUDA"));
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc]: Fluid -> Paddle, we will remove fluid later

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 2]
```diff
@@ -99,18 +102,24 @@ class ArrayToLoDTensorOp : public framework::OperatorBase {
...
-                     i);
-      PADDLE_ENFORCE(x[i].type() == data_type,
-                     "The date type of the %zu'th element in LoDTensorArray "
-                     "differs from previous ones.",
-                     i);
+      PADDLE_ENFORCE_EQ(
+          ins_i_dims, ins_dims,
+          platform::errors::InvalidArgument(
+              "The dimension of the %zu'th element in LoDTensorArray "
+              "differs from previous ones.",
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc]: giving value of `ins_i_dims` and `ins_dims` for debug is better

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 3]
```diff
@@ -99,18 +102,24 @@ class ArrayToLoDTensorOp : public framework::OperatorBase {
...
+              i));
+      PADDLE_ENFORCE_EQ(
+          x[i].place(), place,
+          platform::errors::InvalidArgument(
+              "The place class of the %zu'th element in LoDTensorArray "
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc:115]: place class -> place type? or device?

Comment (chenwhql) [reply paddle/fluid/operators/array_to_lod_tensor_op.cc:115]: and place also can be print later

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 4]
```diff
@@ -99,18 +102,24 @@ class ArrayToLoDTensorOp : public framework::OperatorBase {
...
+              "differs from previous ones.",
+              i));
+      PADDLE_ENFORCE_EQ(
+          x[i].type(), data_type,
+          platform::errors::InvalidArgument(
+              "The date type of the %zu'th element in LoDTensorArray "
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc:122]: give the data type, the current one's data type is *, the previous one's data type is *

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 5]
```diff
@@ -138,7 +147,13 @@ class ArrayToLoDTensorOp : public framework::OperatorBase {
     ArrayToLoDFunctor functor;
     for (size_t idx : table_item_idx) {
       cur_level_lod.push_back(cur_level_lod.back() + table_items[idx].length);
-      PADDLE_ENFORCE_LE(table_items[idx].length, x.size());
+      PADDLE_ENFORCE_LE(table_items[idx].length, x.size(),
+                        platform::errors::InvalidArgument(
+                            "The RankTable items length should less than or "
+                            "equal Input(X) size,"
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc]: `or equal` -> `or equal to`?

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 6]
```diff
@@ -151,7 +166,12 @@ class ArrayToLoDTensorOp : public framework::OperatorBase {
         VLOG(10) << "idx=" << idx << " x_idx=" << x_idx << " ["
                  << ", " << end_offset << "]";
         // Copy data
-        PADDLE_ENFORCE_GE(end_offset, start_offset);
+        PADDLE_ENFORCE_GE(
+            end_offset, start_offset,
+            platform::errors::InvalidArgument(
+                "The lod data start offset should smaller or equal end offset,"
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc]: equal -> equal to

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 7]
```diff
@@ -188,10 +208,12 @@ class ArrayToLoDTensorOpProtoMaker : public framework::OpProtoAndCheckerMaker {
...
-    PADDLE_ENFORCE(context->HasInput("X"),
-                   "ArrayToLoDTensorOp must have input X.");
-    PADDLE_ENFORCE(context->HasInput("RankTable"),
-                   "ArrayToLoDTensorOp must have input RankTable.");
+    PADDLE_ENFORCE_EQ(
+        context->HasInput("X"), true,
+        platform::errors::NotFound("Input(X) of BmmOp should not be null"));
+    PADDLE_ENFORCE_EQ(context->HasInput("RankTable"), true,
+                      platform::errors::NotFound(
+                          "Input(RankTable) of BmmOp should not be null"));
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc]: add `.` in end

File: paddle/fluid/operators/array_to_lod_tensor_op.cc [hunk 8]
```diff
@@ -188,10 +208,12 @@ class ArrayToLoDTensorOpProtoMaker : public framework::OpProtoAndCheckerMaker {
 class ArrayToLoDTensorInferShape : public framework::InferShapeBase {
  public:
   void operator()(framework::InferShapeContext *context) const override {
-    PADDLE_ENFORCE(context->HasInput("X"),
-                   "ArrayToLoDTensorOp must have input X.");
-    PADDLE_ENFORCE(context->HasInput("RankTable"),
-                   "ArrayToLoDTensorOp must have input RankTable.");
+    PADDLE_ENFORCE_EQ(
+        context->HasInput("X"), true,
+        platform::errors::NotFound("Input(X) of BmmOp should not be null"));
```
Comment (chenwhql) [inline paddle/fluid/operators/array_to_lod_tensor_op.cc]: add `.` in end

File: paddle/fluid/operators/lod_tensor_to_array_op.cc [hunk 9]
```diff
@@ -61,7 +61,8 @@ struct LoDTensorToArrayFunctor : public boost::static_visitor<void> {
 #ifdef PADDLE_WITH_CUDA
       Apply(static_cast<platform::CUDADeviceContext *>(dev_ctx));
 #else
-      PADDLE_THROW("Not compiled with cuda");
+      PADDLE_THROW(
+          platform::errors::Unavailable("Fluid is not compiled with CUDA"));
```
Comment (chenwhql) [inline paddle/fluid/operators/lod_tensor_to_array_op.cc]: Fluid -> Paddle

File: paddle/fluid/operators/lod_tensor_to_array_op.cc [hunk 10]
```diff
@@ -109,8 +110,10 @@ class LoDTensorToArrayOp : public framework::OperatorBase {
 
     PADDLE_ENFORCE_LT(
         rank_level, x.lod().size(),
-        "Input should be a LoDTensor, and its lod_level should be at least %d",
-        rank_level + 1);
+        platform::errors::InvalidArgument(
+            "Input should be a LoDTensor, and its lod_level should be at "
+            "least %d",
```
Comment (chenwhql) [inline paddle/fluid/operators/lod_tensor_to_array_op.cc]: add `.` in end

File: paddle/fluid/operators/array_to_lod_tensor_op.cc (hunk 1 above)
Comment (yongqiangma) [reply paddle/fluid/operators/array_to_lod_tensor_op.cc]: Has changed to Paddle

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**